locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```diff
@@ -72,9 +74,17 @@
      */
     private int[] inboundPorts;
 
+    private IpProtocol ipProtocol;
+    private RangeSet<Integer> portRanges;
```

Comment:
I don't like this.  It makes the yaml more complicated.  If you want to merge the tcpPortRanges and udpPortRanges then I'd prefer you change it to accept something like:
portRanges: tcp/22, udp/1000-1100, 8080
where it defaults to tcp when no protocol set.

Otherwise you'd need to created 2 shared security groups - so that you could configure both tcp and udp ports.